src/egon/data/datasets/plotdatascn.py

```diff
@@ -0,0 +1,191 @@
+
+"""
+Tilemapbase library could be necessary
+to generate the figures
+        
+        
+"""
+
+logger = logging.getLogger(__name__)
+
+if 'READTHEDOCS' not in os.environ:
+    from geoalchemy2.shape import to_shape
+
+__copyright__ = ("Flensburg University of Applied Sciences, "
+                 "Europa-Universität Flensburg, "
+                 "Centre for Sustainable Energy Systems, "
+                 "DLR-Institute for Networked Energy Systems")
+__license__ = ""
+__author__ = ""
+
+  
+def set_epsg_network(network):
+    """
+    Change EPSG from 4326 to 3857. Needed when using osm-background.
+
+    Parameters
+    ----------
+    network : PyPSA network container
+
+    Returns
+    -------
+    """
+
+    inProj = Proj(init='epsg:4326')
+    outProj = Proj(init='epsg:3857')
+    x1, y1 = network.buses.x.values, network.buses.y.values
+    x2, y2 = transform(inProj, outProj, x1, y1)
+    network.buses.x, network.buses.y = x2, y2
+    network.epsg = 3857
+    set_epsg_network.counter = set_epsg_network.counter + 1
+    
+def plot_osm(x, y, zoom, alpha=0.4):
+    """
+    Plots openstreetmap as background of network-plots
+
+    Parameters
+    ----------
+    x : array of two floats
+        Define x-axis boundaries (lat) of osm plot
+    y : array of two floats
+        Define y-axis boundaries (long) of osm plot
+    zoom : int
+        Define zoom of osm, higher values for higher resolution
+    alpha : float
+        Sets osm-visibility, increase value if osm covers network-plot
+    osm : bool or dict, e.g. {'x': [1,20], 'y': [47, 56], 'zoom' : 6}
+        If not False, osm is set as background
+        with the following settings as dict:
+                'x': array of two floats, x axis boundaries (lat)
+                'y': array of two floats, y axis boundaries (long)
+                'zoom' : resolution of osm. The default is False.
+
+    Returns
+    -------
+    """
+
+    tilemapbase.init(create=True)
+
+    extent = tilemapbase.Extent.from_lonlat(x[0], x[1], y[0], y[1])
+    extent = extent.to_aspect(1.0)
+    extent = extent.to_project_3857()
+
+    fig, ax = plt.subplots(1,1)
+    ax.set_zorder(1)
+    ax.add_artist(ScaleBar(1))
+    plt.axis('off')
+    plotter = tilemapbase.Plotter(extent, tilemapbase.tiles.build_OSM(),
+                                  zoom=zoom)
+    plotter.plot(ax, alpha=alpha)
+    #ax.plot(x, y, "ro-")
+    return fig, ax
+
+def plot_generation(
+              carrier,scenario, osm = False
+            ):
+   """
+    Plots color maps according to the capacity of different generators
+    of the two existing scenarios (eGon2035 and eGon100RE)
+    
+
+    Parameters
+    ----------
+    carrier : generators
+    The list of generators: biomass, central_biomass_CHP, central_biomass_CHP_heat,
+    industrial_biomass_CHP, solar, solar_rooftop, wind_offshore, wind_onshore. 
+      
+    scenario: eGon2035, eGon100RE
+        
+        
+   """
+    
+
+   con = db.engine()
```

Comment:
Please include a note explaining that this function must be called while in the folder that contains the file egon-data.configuration.yaml.

Reply:
done